### PR TITLE
Don't trigger onClick if mouse moved beyond distance

### PR DESCRIPTION
**Pull request by vidaaudrey**
## Problem 
The Worldview `onClick` handler is being fired even when user is panning around because currently `onClicked` is always fired after `onMouseUp`. 

## Fix 
Track the movement, only fire `onClick` if the mouse has moved a distance within 3 screen pixels. 

## Test
Manually test: press and pan around for long distance, the information display should not change. Will add a drag object demo later in a separate PR. 

demo

### CODE REVIEW

File: packages/regl-worldview/src/Worldview.js [hunk 1]
```diff
@@ -165,6 +176,16 @@ export class WorldviewBase extends React.Component<BaseProps, State> {
...
+    const { dragStartPos } = this.state;
+    if (dragStartPos) {
+      const deltaX = e.clientX - dragStartPos.x;
+      const deltaY = e.clientY - dragStartPos.y;
+      const distance = Math.sqrt(deltaX * deltaX + deltaY * deltaY);
+      if (distance > this.props.mouseClickRadius) {
+        this._disableOnClick = true;
+      }
+      this.setState({ dragStartPos: null });
+    }
```
Comment (janpaul123) [inline packages/regl-worldview/src/Worldview.js]: If you want to simplify things a bit, you can even just call `onClick` here (if the mouse has not moved beyond the radius). Then you can get rid of `_disableOnClick`.

File: packages/regl-worldview/src/Worldview.js [hunk 2]
```diff
@@ -104,6 +108,7 @@ export class WorldviewBase extends React.Component<BaseProps, State> {
         cameraState: props.cameraState || props.defaultCameraState || DEFAULT_CAMERA_STATE,
         onCameraStateChange: props.onCameraStateChange || undefined,
       }),
+      dragStartPos: null,
```
Comment (robin-pham) [inline packages/regl-worldview/src/Worldview.js]: we don't have to put this in the component `state` if it's not going to be used in the `render` method. Let's just keep it as a class property instead.

File: docs/src/3.1.Worldview.mdx [hunk 3]
```diff
@@ -2,22 +2,23 @@
...
+| `hitmapOnMouseMove`   | `boolean`                                       |                        | if enabled, the clicked objectId will be returned from the mouse event handler                                         |
+| `children`            | `React.Node`                                    |                        | regl components to render inside the Worldview                                                                         |
+| `style`               | `{ [styleAttribute: string]: number | string }` | {}                     | html style object for the Worldview root element                                                                       |
+| `showDebug`           | `boolean`                                       |                        | show debug information                                                                                                 |
+| `mouseClickRadius`    | `number`                                        | 3                      | if the mouse has moved distance beyond the mouseClickRadius limit, the click event handler isn't going to be triggered |
```
Comment (robin-pham) [inline docs/src/3.1.Worldview.mdx]: Should we set it `Number.MAX_SAFE_INTEGER` by default, as this behavior would be more consistent with the DOM? Or should we just set sensible defaults and people will have to opt-out of this behavior?

File: docs/src/3.4.MouseEvents.mdx [hunk 4]
```diff
@@ -75,3 +75,7 @@ Check out below example for automatic/custom hitmap mapping, and regular/instanc
 ## Full Example
 
 <MouseEvents />
+
+## Other
+
+We use `mouseClickRadius` prop to determine if mouseClick event is going to be triggered, specificially if the mouse has moved beyond the `mouseClickRadius` limit, the `onClick` event handler will be canceled, and `onMouseUp` will still get triggered if present.
```
Comment (robin-pham) [inline docs/src/3.4.MouseEvents.mdx]: probably best to also add a description of the common use case, i.e. this helps implement the desired behavior of dragging the camera around not firing `onClick` events

File: packages/regl-worldview/src/Worldview.js [hunk 5]
```diff
@@ -149,14 +154,20 @@ export class WorldviewBase extends React.Component<BaseProps, State> {
...
   };
 
   _onDoubleClick = (e: MouseEvent) => {
+    this._disableOnClick = false;
     this._onMouseInteraction(e, "onDoubleClick");
   };
 
   _onMouseDown = (e: MouseEvent) => {
+    this._disableOnClick = false;
+    this.setState({ dragStartPos: { x: e.clientX, y: e.clientY } });
```
Comment (brianc) [inline packages/regl-worldview/src/Worldview.js]: instead of using state for this what about just setting it as an instance variable on the component? Will remove a render & draw call that I don't think we need.

File: docs/src/3.1.Worldview.mdx (hunk 3 above)
Comment (brianc) [inline docs/src/3.1.Worldview.mdx]: might be nice to mention in the docs the number is number of pixels in screen space.  There's a very small potential one could think it means the distance in the scene.

File: packages/regl-worldview/src/Worldview.js (hunk 5 above)
Comment (vidaaudrey) [reply packages/regl-worldview/src/Worldview.js]: Good idea! Will update

File: docs/src/3.4.MouseEvents.mdx (hunk 4 above)
Comment (vidaaudrey) [reply docs/src/3.4.MouseEvents.mdx]: Good idea! Will add

File: packages/regl-worldview/src/Worldview.js (hunk 1 above)
Comment (vidaaudrey) [reply packages/regl-worldview/src/Worldview.js]: Good suggestion. Updated

File: docs/src/3.1.Worldview.mdx (hunk 3 above)
Comment (vidaaudrey) [reply docs/src/3.1.Worldview.mdx]: As discussed offline, DOM onClick doesn't get triggered if we are dragging objects around.  Here we detect dragging by the moving distance which is somewhat close to what we want to achieve.

File: packages/regl-worldview/src/Worldview.js [hunk 6]
```diff
@@ -32,7 +33,7 @@ export type BaseProps = {|
   cameraState?: CameraState,
   onCameraStateChange?: (CameraState) => void,
   defaultCameraState?: CameraState,
-
+  mouseClickRadius: number,
```
Comment (janpaul123) [inline packages/regl-worldview/src/Worldview.js]: I'm not sure if we should support this as a property or just hardcode it. The fewer properties the better, because each one of these we will have to support indefinitely (until we bump major versions).

Comment (janpaul123) [reply packages/regl-worldview/src/Worldview.js]: (And I don't see any good reason for people to change this property?)

Comment (vidaaudrey) [reply packages/regl-worldview/src/Worldview.js]: You are right. The fewer props the better. Just moved.

File: docs/src/jsx/utils/WorldviewCodeEditor.js [hunk 7]
```diff
@@ -113,7 +114,7 @@ export const scope = {
   Text,
   GLTFScene,
   withPose,
-
+  getRayFromClick,
```
Comment (janpaul123) [inline docs/src/jsx/utils/WorldviewCodeEditor.js]: This change seems unrelated to the rest of this PR, did you mean to commit this?

File: docs/src/3.4.MouseEvents.mdx [hunk 8]
```diff
@@ -75,3 +75,7 @@ Check out below example for automatic/custom hitmap mapping, and regular/instanc
 ## Full Example
 
 <MouseEvents />
+
+## Other
+
+We use `DEFAULT_MOUSE_CLICK_RADIUS` (3 pixels in the screen space) to determine if mouse click event is going to be triggered, specificially if the mouse has moved beyond the `DEFAULT_MOUSE_CLICK_RADIUS` limit, the `onClick` event handler will be canceled, and `onMouseUp` will still get triggered if present. This helps implement certain behaviors such as panning around worldview without firing `onClick` event.
```
Comment (robin-pham) [inline docs/src/3.4.MouseEvents.mdx]: we don't really need to know that the variable is called `DEFAULT_MOUSE_CLICK_RADIUS`. Either we don't mention anything about this, or we should have a small blurb e.g. "`onClick` events don't fire if the user moves their cursor more than 3 pixels between mouse down and mouse up"

File: docs/src/jsx/utils/WorldviewCodeEditor.js (hunk 7 above)
Comment (vidaaudrey) [reply docs/src/jsx/utils/WorldviewCodeEditor.js]: I started out hoping to build the dragging example, but realized it's better to keep it in a separate PR. Overall it's good to have this function in the global scope of code playground just people can use it.

File: docs/src/3.4.MouseEvents.mdx (hunk 8 above)
Comment (vidaaudrey) [reply docs/src/3.4.MouseEvents.mdx]: Updated: `onClick` events don't fire if the user moves their cursor for more than 3 screen pixels between mouse down and mouse up. This helps implement certain behaviors such as panning around worldview without firing `onClick` event.
